Source/Application.js

```diff
@@ -0,0 +1,291 @@
+// ********************************
+// # Application.js
+// # Sketch Javascript API.
+//
+// All code (C) 2016 Bohemian Coding.
+// ********************************
+
+// ## Imports
+
+import { WrappedObject } from './WrappedObject.js'
+import { Document } from './Document.js'
+import { Rectangle } from './Rectangle.js'
+import { Group } from './Group.js'
+import { Text } from './Text.js'
+import { Image } from './Image.js'
+import { Shape } from './Shape.js'
+import { Artboard } from './Artboard.js'
+import { Page } from './Page.js'
+
+/**
+    Gives you access to Sketch, and provides access to:
+    - the document model and the layer tree
+    - metadata abound sketch itself
+    - utilities for interacting with the user
+    - access to the running plugin, it's resources and settings
+*/
+
+export class Application extends WrappedObject {
+
+    /**
+        Construct a new Application object.
+        An instance of this class will be passed back to you when you
+        initialise the API, so you generally shouldn't need to make new ones.
+
+        @param context The context dictionary passed to the script when it was invoked.
+        @return A new Application object.
+    */
+    
+    constructor(context) {
+      super(context)
+      this.metadata = MSApplicationMetadata.metadata()
+
+      // this is a slightly clumsy way to pass related classes into Layer
+      // without setting up a circular dependency between classes (eg Layer imports Artboard which imports Layer...)
+      // there has to be a better way to do this...
+      this.factory = {
+        "Group" : Group,
+        "Page" : Page,
+        "Artboard" : Artboard,
+        "Shape" : Shape,
+        "Image" : Image,
+        "Text" : Text
+      }
+    }
+
+    /**
+        The version of this API.
+
+        @return A version string.
+    */
+
+    get api_version() {
+      return "1.1"
+    }
+
+    /**
+        The context that the API was created in.
+        This is the traditional context argument that is
+        passed to all plugin scripts when they are run.
+
+        In general you should use the API to access Sketch
+        rather than interacting with the context directly.
+
+        @return A context dictionary.
+    */
+
+    get context() {
+      return this.object
+    }
+
+
+    /**
+        The version of Sketch that is running.
+
+        @return The version as a string, eg "3.6".
+    */
+
+    get version() {
+      return this.metadata['appVersion']
+    }
+
+    /**
+        The exact build of Sketch that is running.
+
+        @return The build number as a string, eg "15352".
+    */
+
+    get build() {
+      return this.metadata['build']
+    }
+
+    /**
+        Returns the full version of Sketch that is running
+
+        @return {string} Version and build number as a string, eg "3.6 (15352)".
```

Comment:
And now I understand how the documenting is supposed to work, so disregard the comments I made in https://github.com/BohemianCoding/Sketch/pull/9822